File: nodejs/muse/data_structures/LinkedStackTest.js
"use strict"

const LinkedStack = require("muse/data_structures/LinkedStack")
const TestRunner  = require("muse/util/TestRunner")

const testLinkedStack = () => {
    let size = 8192

    const stack = new LinkedStack.LinkedStack()

    for (let i = 1; i <= size; i++) {
        stack.push(i)
    }

    if (stack.size() !== size) {
        return false
    }

    for (let i = size; i > 0; i--) {
        if (stack.peek() !== i) {
            return false
        }
        stack.pop()
    }

    return stack.isEmpty()
}

if (module === require.main) {
    TestRunner.pick(testLinkedStack)
}

File: ruby/muse/data_structures/LinkedStackTest.rb
require "muse/data_structures/LinkedStack"
require "muse/util/TestRunner"

def testLinkedStack
  size = 8192

  stack = LinkedStack::LinkedStack.new

  1.upto(size) do |i|
    stack.push(i)
  end

  return false if stack.size != size

  size.downto(1) do |i|
    return false if stack.peek != i

    stack.pop
  end

  stack.isEmpty
end

TestRunner.pick(testLinkedStack) if __FILE__ == $PROGRAM_NAME
